web/modules/custom/rdf_entity/modules/rdf_file/src/Plugin/Field/FieldType/RdfFileItem.php

```diff
@@ -0,0 +1,364 @@
+<?php
+
+namespace Drupal\rdf_file\Plugin\Field\FieldType;
+
+use Drupal\Component\Utility\Bytes;
+use Drupal\Component\Render\PlainTextOutput;
+use Drupal\Component\Utility\Random;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StreamWrapper\StreamWrapperInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\DataReferenceTargetDefinition;
+
+// Constraints = {"ReferenceAccess" = {}, "FileValidation" = {}}.
+/**
+ * Plugin implementation of the 'file' field type.
+ *
+ * @FieldType(
+ *   id = "rdf_file",
+ *   label = @Translation("RDF File"),
+ *   description = @Translation("This field stores the ID of a file as an URI."),
+ *   category = @Translation("Reference"),
+ *   default_widget = "rdf_file_generic",
+ *   default_formatter = "rdf_file_default",
+ *   list_class = "\Drupal\rdf_file\Plugin\Field\FieldType\RdfFileFieldItemList",
+ *
+ * )
+ */
+class RdfFileItem extends EntityReferenceItem {
```

Comment:
I don't understand why you didn't start directly from FileItem as we already have the whole file functionality there. Then, in RdfFileItem, you simply extend `\Drupal\Core\Field\FieldItemInterface::schema()`:

```php
public static function schema(FieldStorageDefinitionInterface $field_definition) {
  $schema = parent::schema();
  $schema['fields']['uri'] = [
     'description' => 'The internal or external file URI.',
     'type' => 'varchar',
     'length' => 512,
  ];
  return $schema;
}
```

Actually a our field type is a file field having an additional column/property. As an edge implementation we can just use the standard `description` that already exists in FileItem and store there the URI.

The we extend the file widget and formatter to use this property.